File: examples/nodejs/basic-conversion.js
/**
 * Basic Image Conversion Example
 *
 * This example shows how to convert a single image file
 */

const ConvertorioClient = require('../../libs/nodejs');

// Initialize the client with your API key
const client = new ConvertorioClient({
    apiKey: 'your_api_key_here' // Get your API key from https://convertorio.com/account
});

async function main() {
    try {
        console.log('Starting image conversion...\n');

        // Convert an image
        const result = await client.convertFile({
            inputPath: './input.png',
            targetFormat: 'jpg'
            // outputPath is optional - will auto-generate if not provided
        });

        console.log('\n✓ Conversion completed successfully!');
        console.log('Result:', {
            inputFile: result.inputPath,
            outputFile: result.outputPath,
            format: `${result.sourceFormat} → ${result.targetFormat}`,
            fileSize: `${(result.fileSize / 1024).toFixed(2)} KB`,
            processingTime: `${result.processingTime} ms`
        });

    } catch (error) {
        console.error('✗ Conversion failed:', error.message);
        process.exit(1);
    }
}

main();

File: examples/ruby/basic_conversion.rb
#!/usr/bin/env ruby

# Basic Image Conversion Example
#
# This example shows how to convert a single image file

require_relative '../../libs/ruby/lib/convertorio'

# Initialize the client with your API key
client = Convertorio::Client.new(
  api_key: 'your_api_key_here' # Get your API key from https://convertorio.com/account
)

begin
  puts "Starting image conversion...\n\n"

  # Convert an image
  result = client.convert_file(
    input_path: './input.png',
    target_format: 'jpg'
    # output_path is optional - will auto-generate if not provided
  )

  puts "\n✓ Conversion completed successfully!"
  puts "Result:"
  puts "  Input file: #{result[:input_path]}"
  puts "  Output file: #{result[:output_path]}"
  puts "  Format: #{result[:source_format]} → #{result[:target_format]}"
  puts "  File size: #{'%.2f' % (result[:file_size] / 1024.0)} KB"
  puts "  Processing time: #{result[:processing_time]} ms"

rescue Convertorio::Error => e
  puts "✗ Conversion failed: #{e.message}"
  exit(1)
end
